crates/arroyo-connectors/src/iggy/mod.rs

```diff
@@ -0,0 +1,624 @@
+                    _transport: profile.transport.to_string(),
+                    username,
+                    password,
+                    offset_mode,
+                    auto_commit: *auto_commit,
+                    format: config.format.expect("Format must be set for Iggy source"),
+                    framing: config.framing,
+                    bad_data: config.bad_data,
+                    client: None,
+                    messages_per_second: std::num::NonZeroU32::new(
+                        config
+                            .rate_limit
+                            .map(|l| l.messages_per_second)
+                            .unwrap_or(u32::MAX),
+                    )
+                    .unwrap(),
+                    metadata_fields: config.metadata_fields,
+                })))
+            }
+            TableType::Sink {
+                partitioning,
+                partition_id,
+            } => {
+                let partitioning_strategy = match partitioning {
+                    SinkPartitioning::PartitionId => {
+                        if let Some(id) = partition_id {
+                            PartitioningStrategy::PartitionId((*id) as u32)
+                        } else {
+                            bail!("Partition ID is required for partition_id partitioning strategy")
+                        }
+                    }
+                    SinkPartitioning::Balanced => PartitioningStrategy::Balanced,
+                };
+
+                let (username, password) = match &profile.authentication {
+                    IggyConfigAuthentication::None {} => (None, None),
+                    IggyConfigAuthentication::UsernamePassword { username, password } => (
+                        Some(username.clone()),
+                        Some(
+                            password
+                                .sub_env_vars()
+                                .expect("Failed to substitute env vars"),
+                        ),
+                    ),
+                };
+
+                Ok(ConstructedOperator::from_operator(Box::new(IggySinkFunc {
+                    stream: table.stream.clone(),
+                    topic: table.topic.clone(),
+                    partitioning: partitioning_strategy,
+                    endpoint: profile.endpoint.clone(),
+                    _transport: profile.transport.to_string(),
+                    username,
+                    password,
+                    client: None,
+                    serializer: ArrowSerializer::new(
+                        config.format.expect("Format must be defined for IggySink"),
+                    ),
+                })))
+            }
+        }
+    }
+}
+
+pub struct IggyTester {
+    pub connection: IggyConfig,
+}
+
+impl IggyTester {
+    async fn connect(&self) -> Result<IggyClient, String> {
+        info!("Testing connection to Iggy at {}", self.connection.endpoint);
+
+        let client = IggyClient::default();
```

Comment:
It looks like we're not actually using the endpoint, so it will only work with the default